### ISSUE TITLE
Unable to run `npx inngest-cli@latest dev` on Windows 11

### ISSUE THREAD

**Issue by dmitryyankowski** [inngest/inngest#449]
I'm trying to start the inngest-cli dev on a Windows 11 computer. However when it tries to start I get the following error:

`
This version of C:\Users\Admin\AppData\Local\npm-cache\_npx\84e0b49cabd4122d\node_modules\inngest-cli\bin\inngest.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.
`

I was able to run it just fine a few months back, but am now getting this error. Any ideas? 😸

**Comment by jpwilliams** [comment]
@iliketowritecode Working okay on my Windows 11 (OS build 22621.1702). Could you let us know what version you're running on?

Should be able to find it via Start Menu > "About your PC".

**Comment by AlinaMos** [comment]
I have the same problem on my Windows 11- 22621.1702

**Comment by tonyhb** [comment]
🤔 okay, taking a look.  i'm wondering what's happening here;  we're using Go and the only changes we've made are to update to Go 1.20, in terms of our build.

I'll try and repro this.  This is latest, so let me validate what's up on a VM.

**Comment by brenden-js** [comment]
Just updated windows 10 to 11, tried uninstalling and reinstalling globally with both npm and pnpm which was a fix in the discord but still getting the same error:

```
C:\...\Desktop>npx inngest-cli@latest dev
This version of C:\...\node_modules\inngest-cli\bin\inngest.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.

```

**Comment by brenden-js** [comment]
Okay update, I got it working on my machine. For some reason npm/pnpm was using the windows ARM 64 exe (Im on an x64 machine so the not compatible error makes sense), so I manually downloaded the x86/64 exe from here:

(https://github.com/inngest/inngest/releases/tag/v0.13.6)

Then replaced the inngest.exe version in this folder...:

`C:\...\AppData\Local\npm-cache\_npx\----some random hash-----\node_modules\inngest-cli\bin
`
...with the x86/64 one from the _inngest_0.13.6_windows_x86_64.zip_ from the above page.

Probably a really ugly workaround but at least I can test locally again lol. Not sure if it's an error in the setup/config on my machine or what, but hopefully this helps!

**Comment by brenden-js** [comment]
Okay update, I got it working on my machine. For some reason npm/pnpm was using the windows ARM 64 inngest.exe (I'm on an x64 machine so the "Windows not compatible error" makes sense), so I manually downloaded the x86/64 inngest.exe from here:

(https://github.com/inngest/inngest/releases/tag/v0.13.6)

Then installed  inngest-cli@latest globally and replaced the inngest.exe version in the bin folder with the correct x86/64 version:

`X:\AppData\Roaming\npm\node_modules\inngest-cli\bin`

I don't know if this was caused by something on my end with npm config or something, but hopefully this helps.

**Comment by glitchyg** [comment]
I posted about the same issue about a month ago in discord. Seems something strange is going on with the installer picking the incorrect version
https://discord.com/channels/842170679536517141/1103400555902144573

**Comment by jpwilliams** [comment]
We use [`process.arch`](https://nodejs.org/api/process.html#processarch) to determine the system architecture, then try and match that to our two supported architectures: [x86_64 and arm64](https://github.com/inngest/inngest/blob/main/npm/postinstall.ts#L10-L13).

Could someone facing the issue run a couple of commands? In this case I'd expect that the first might be returning `arm64` for you, though I also feel that's highly unlikey. 🤔

```sh
node -e 'console.log(process.arch)'
```

```sh
npx envinfo --system
```

**Comment by brenden-js** [comment]
For me the 'node -p 'process.arch' command was just returning "process.arch" for some reason:
```
node -p 'process.arch'
process.arch
```

But doing this command shows something useful: 
```
node -e console.log(process.arch)
x64
```

and the second command is showing

```
npx envinfo --system

  System:
    OS: Windows XX 1X.XX.XX.X.X
    CPU: (16) x64 Intel(R) Core(TM) iX-XXX-XX CPU @ X.XGHz
    Memory: X GB / X GB

```

So the system says its x64, but for some reason it was installing the ARM version. Strange.

**Comment by jpwilliams** [comment]
Hmm yeah, that all looks pretty expected. Thank you for running those!

The docs for `process.arch` state that it's the archictecture "_for which the Node.js binary was compiled_," but it seems super unlikely to me that you'd somehow have installed an arm64 Node binary _that also somehow runs_ which then causes the problem.

The water's a bit muddy here. This user reports that `process.arch` changes values depending on what's _invoking_ it, which doesn't really match the documentation's description of the property. 😬
[javascript - "process.arch" returns wrong value when Node script is run through Ruby - Stack Overflow](https://stackoverflow.com/questions/73126075/process-arch-returns-wrong-value-when-node-script-is-run-through-ruby)

Are you using a less common shell or maybe a particular tool for managing Node versions? I think `envinfo` can give a clue on the Node version management side with `npx envinfo --binaries`.

**Comment by iliketowritecode** [comment]
Here are my outputs for both of those commands:

`npx envinfo --system`
```
System:
    OS: Windows 10 10.0.22621
    CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
    Memory: 43.00 GB / 63.91 GB
```

`npx envinfo --binaries`
```
 Binaries:
    Node: 20.2.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.6.6 - C:\Program Files\nodejs\npm.CMD
```

I installed NodeJS via the normal Windows NodeJS installer.

**Comment by hgogoi2012** [comment]
When this issue will be fixed ? I am unable to install it in my windows 11

**Comment by charanjit-singh** [comment]
Hey guys, I know this is not a right solution, but for now, on my local windows machine I'm able to run it using docker.

```
    "run-inngest": "docker build -f ./docker/ingest.Dockerfile . -t inngest-cli && docker run --publish 8288:8288 inngest-cli",
```

Here's my dockerfile
```
FROM node:latest

RUN npm install -g inngest-cli@latest
RUN npm install -g http-server
RUN npm install -g concurrently


CMD ["concurrently","inngest-cli dev -u http://localhost:3000/api/inngest", "http-server -p 3000 -P http://host.docker.internal:3000 --silent"]

# Expose port 8288
EXPOSE 8288
```


For now, I'm able to run inngest in windows using docker, with reverse proxy server in the container to connect to host.

**Comment by jpwilliams** [comment]
This issue should now be resolved in [v0.16.4](https://github.com/inngest/inngest/releases/tag/v0.16.4) following #575.

Windows builds for >v0.14 are still being worked on in #521 and #550, so `npx inngest-cli@latest dev` on Windows will now use the latest supported version for that platform, [v0.14.6](https://github.com/inngest/inngest/releases/tag/v0.14.6).

**Comment by JCMisa** [comment]
why my inngest inside X:\AppData\Roaming\npm\node_modules\inngest-cli\bin a File type. When i run inngest --version it only opens the file path instead of a url.
